tools/visualizations/vis_pipeline.py

```diff
@@ -143,43 +148,80 @@ def build_dataset_pipeline(cfg, phase):
     origin_pipeline = data_cfg.pipeline
     data_cfg.pipeline = loadimage_pipeline
     dataset = build_dataset(data_cfg)
-    pipeline = Compose(origin_pipeline)
-
-    return dataset, pipeline
-
-
-def put_img(board, img, center):
-    """put a image into a big board image with the anchor center."""
-    center_x, center_y = center
-    img_h, img_w, _ = img.shape
-    xmin, ymin = int(center_x - img_w // 2), int(center_y - img_h // 2)
-    board[ymin:ymin + img_h, xmin:xmin + img_w, :] = img
+    pipelines = {
+        pipeline_cfg['type']: build_from_cfg(pipeline_cfg, PIPELINES)
+        for pipeline_cfg in origin_pipeline
+    }
+
+    return dataset, pipelines
+
+
+def concat_imgs(args, imgs: List[np.ndarray], steps: List[str]) -> np.ndarray:
+    """Concat list of pictures into a single big picture, align height here."""
+    ori_shapes = [img.shape for img in imgs]
+    # if src image is too large or too small, it will make the text
+    # information difficult to confirm, so here add adaptive adjustment.
+    if args.adaptive:
```

Comment:
Can add a warning about suggestion of using `--adaptive` for small images.

Reply:
Do you mean when reading some pictures, which have very small or very lager resolution, then we give a warning like
"the src picture have size of xxx, it is better to add `--adaptive` to adaptively resize to show?"

Reply:
done.